src/System.Management.Automation/utils/ClrFacade.cs

```diff
@@ -497,55 +482,12 @@ internal static string ToDmtfDateTime(DateTime date)
         }
 
         /// <summary>
-        /// Facade for FormatterServices.GetUninitializedObject.
-        ///
-        /// In CORECLR, there are two peculiarities with its implementation that affect our own:
-        /// 1. Structures cannot be instantiated using GetConstructor, so they must be filtered out.
-        /// 2. Classes must have a default constructor implemented for GetConstructor to work.
-        ///
-        /// See RemoteHostEncoder.IsEncodingAllowedForClassOrStruct for a list of the required types.
-        /// </summary>
-        /// <param name="type"></param>
-        /// <returns></returns>
-        internal static object GetUninitializedObject(Type type)
-        {
-#if CORECLR
-            switch (type.Name)
-            {
-                case "KeyInfo"://typeof(KeyInfo).Name:
-                    return new KeyInfo(0, ' ', ControlKeyStates.RightAltPressed, false);
-                case "Coordinates"://typeof(Coordinates).Name:
-                    return new Coordinates(0, 0);
-                case "Size"://typeof(Size).Name:
-                    return new Size(0, 0);
-                case "BufferCell"://typeof(BufferCell).Name:
-                    return new BufferCell(' ', ConsoleColor.Black, ConsoleColor.Black, BufferCellType.Complete);
-                case "Rectangle"://typeof(Rectangle).Name:
-                    return new Rectangle(0, 0, 0, 0);
-                default:
-                    ConstructorInfo constructorInfoObj = type.GetConstructor(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null, new Type[] { }, null);
-                    if (constructorInfoObj != null)
-                    {
-                        return constructorInfoObj.Invoke(new object[] { });
-                    }
-                    return new object();
-            }
-#else
-            return FormatterServices.GetUninitializedObject(type);
-#endif
-        }
-
-        /// <summary>
         /// Facade for ProfileOptimization.SetProfileRoot
         /// </summary>
         /// <param name="directoryPath">The full path to the folder where profile files are stored for the current application domain.</param>
         internal static void SetProfileOptimizationRoot(string directoryPath)
         {
-#if CORECLR
             PSAssemblyLoadContext.SetProfileOptimizationRootImpl(directoryPath);
```

Comment:
Yes we should, and it requires corresponding changes in `CorePsAssemblyLoadContext.cs` as `SetProfileOptimizationRootImpl` is no longer needed. I will do it in subsequent PRs.